src/PubSubClient.cpp

```diff
@@ -212,21 +212,18 @@ bool PubSubClient::connect(const char* id, const char* user, const char* pass, c
 }
 
 bool PubSubClient::connected() {
-    bool rc;
-    if (_client == NULL) {
-        rc = false;
-    } else {
-        rc = (bool)_client->connected();
-        if (!rc) {
-            if (_state == MQTT_CONNECTED) {
-                DEBUG_PSC_PRINTF("lost connection (client may have more details)\n");
-                _state = MQTT_CONNECTION_LOST;
-                _client->flush();
-                _client->stop();
-            }
-        } else {
-            return _state == MQTT_CONNECTED;
+    if (!_client) return false;
+
+    bool rc = (bool)_client->connected();
```

Comment:
No need to cast to `bool`, especially not using a C-style cast.

Also the bool can be completely eliminated here as in the else scope (with `rc == true`) the function will immediately return.

So can be `if (_client->connected()) return _state == MQTT_CONNECTED;`
then the rest of what to do when `_state == MQTT_CONNECTED` and `return false;` at the end.